src/main/kotlin/org/rust/ide/intentions/ConvertClosureToFunctionIntention.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ref.MethodResolveVariant
+
+class ConvertClosureToFunctionIntention : RsElementBaseIntentionAction<ConvertClosureToFunctionIntention.Context>() {
+
+    override fun getText(): String = "Convert closure to function"
+    override fun getFamilyName(): String = "Convert between local function and closure"
+
+    data class Context(
+        val assignment: RsLetDecl,
+        val lambda: RsLambdaExpr,
+    )
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        // We try to find a let declaration
+        val possibleTarget = element.ancestorStrict<RsLetDecl>() ?: return null
+
+        // we cannot convert lambdas bound to "_"
```

Comment:
Maybe a better approach would be to create the function always (even if there is an invalid binding or no binding at all), and let the user change it's name afterwards. If the binding was invalid, the default name of the function could be something autogenerated (like `fn0`).

`RsInPlaceVariableIntroducer` or `buildAndRunTemplate` can be used for the renaming.

Reply:
I like that idea. I changed it so it builds the template in case an invalid function name would be generated. Please tell me if there's a better way for that than `isValidVariableIdentifier()`. Also, should we change the availability of this intention to only the arguments and maybe the variable name as well? Currently, this intention would also be available anywhere in the closure body.

Reply:
About the availability: yes, I suggest to only offer it in the signature of functions and parameter list + variable name for closures (https://github.com/intellij-rust/intellij-rust/pull/6813#discussion_r571641678). I think that `isValidVariableIdentifier` is ok.